File: 35.cpp
// https://leetcode.com/problems/search-insert-position/
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        auto itpair = std::equal_range(std::begin(nums), std::end(nums), target);
        if (itpair.first == std::end(nums) || *itpair.first == target) {
            return std::distance(std::begin(nums), itpair.first);
        } else {
            return std::distance(std::begin(nums), itpair.second);
        }
    }
};

File: golang/35.go
package main

func searchInsert(nums []int, target int) int {
	l := len(nums)
	idx1 := sort.Search(l, func(i int) bool { return nums[i] >= target })
	idx2 := sort.Search(l, func(i int) bool { return nums[i] > target })

	if idx1 == l || nums[idx1] == target {
		return idx1
	} else {
		return idx2
	}
}
